Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -0,0 +1,1114 @@
+                                         or flag_info_face_y(i + i_n, j, k + k_n) == 2)
+                                    and flag_ext_face_y(i, j, k)) {
+                                    Sy_mod(i + i_n, j, k + k_n) -= Sy_ext;
+                                    // Insert the index of the face info
+                                    *(borrowing_y_inds + ps) = ps;
+                                    // Store the information about the intruded face in the dataset of the
+                                    // faces which are borrowing area
+                                    FaceInfoBox::addConnectedNeighbor(i_n, k_n, ps,
+                                                                      borrowing_y_neigh_faces);
+                                    *(borrowing_y_area + ps) = Sy_ext;
+
+                                    flag_info_face_y(i + i_n, j, k + k_n) = 2;
+                                    // Add the area to the intruding face.
+                                    Sy_mod(i, j, k) = Sy(i, j, k) + Sy_ext;
+                                    flag_ext_face_y(i, j, k) = false;
+                                }
+                            }
+                        }
+                    }
+                }
+            }, amrex::Scan::Type::exclusive);
+
+        borrowing_y.inds.resize(nelems_y);
+
+    }
+    // Do the extensions in the z-plane
+    idim = 2;
+
+    for (amrex::MFIter mfi(*Bfield_fp[maxLevel()][idim]); mfi.isValid(); ++mfi) {
+
+        amrex::Box const &box = mfi.validbox();
+
+        auto const &Sz = m_face_areas[maxLevel()][idim]->array(mfi);
+        auto const &flag_ext_face_z = m_flag_ext_face[maxLevel()][idim]->array(mfi);
+        auto const &flag_info_face_z = m_flag_info_face[maxLevel()][idim]->array(mfi);
+        auto &borrowing_z = (*m_borrowing[maxLevel()][idim])[mfi];
+        auto const &borrowing_z_inds_pointer = borrowing_z.inds_pointer.array();
+        auto const &borrowing_z_size = borrowing_z.size.array();
+        amrex::Long ncells = box.numPts();
+        int* borrowing_z_inds = borrowing_z.inds.data();
+        uint8_t* borrowing_z_neigh_faces = borrowing_z.neigh_faces.data();
+        double* borrowing_z_area = borrowing_z.area.data();
+
+        auto const &Sz_mod = m_area_mod[maxLevel()][idim]->array(mfi);
+        const auto &lx = m_edge_lengths[maxLevel()][0]->array(mfi);
+        const auto &ly = m_edge_lengths[maxLevel()][1]->array(mfi);
+        amrex::Real dx = cell_size[0];
+        amrex::Real dy = cell_size[1];
+
+
+        nelems_z = amrex::Scan::PrefixSum<int>(ncells,
+            [=] AMREX_GPU_DEVICE (int icell) {
+                amrex::Dim3 cell = box.atOffset(icell).dim3();
+                int i = cell.x;
+                int j = cell.y;
+                int k = cell.z;
+                // If the face doesn't need to be extended break the loop
+                if (!flag_ext_face_z(i, j, k)) {
+                    //borrowing_z_size(i, j, k) = 0;
```

Comment:
Just double checking if this line has been commented out on purpose. If so, shall we remove it?